askbot/models/__init__.py

```diff
@@ -4109,6 +4113,11 @@ def record_spam_rejection(
     post_anonymous_askbot_content,
     dispatch_uid='post_anon_content_on_login'
 )
+signals.post_save.connect(
```

Comment:
Causes test cases to fail. Sounds good in principle, but part of me thinks that it might be good to call reset_cached_data explicitly where needed.